playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/provider/PlayBillingWrapper.java

```diff
@@ -110,18 +110,21 @@ public void consume(String token, ConsumeResponseListener callback) {
 
     @Override
     public boolean launchPaymentFlow(Activity activity, SkuDetails sku, MethodData methodData) {
-        BillingFlowParams.Builder builder = BillingFlowParams.newBuilder();
-
-        builder.setSkuDetails(sku);
+        BillingFlowParams.SubscriptionUpdateParams.Builder subUpdateParamsBuilder = BillingFlowParams.SubscriptionUpdateParams.newBuilder();
 
-        if (methodData.oldSku != null && methodData.purchaseToken != null) {
-            builder.setOldSku(methodData.oldSku, methodData.purchaseToken);
+        if (methodData.purchaseToken != null) {
+            subUpdateParamsBuilder.setOldSkuPurchaseToken(methodData.purchaseToken);
         }
 
         if (methodData.prorationMode != null) {
-            builder.setReplaceSkusProrationMode(methodData.prorationMode);
+            subUpdateParamsBuilder.setReplaceSkusProrationMode(methodData.prorationMode);
         }
 
+        BillingFlowParams.Builder builder = BillingFlowParams.newBuilder();
+
+        builder.setSkuDetails(sku);
+        builder.setSubscriptionUpdateParams(subUpdateParamsBuilder.build());
```

Comment:
Do you mind rejigging this code so that we don't call `setSubscriptionUpdateParams` unless the old purchase token or proration mode is set?

Reply:
done! Sorry for the delay.